ingestion-beam/src/main/java/com/mozilla/telemetry/transforms/PubsubMessageToTableRow.java

```diff
@@ -252,21 +252,25 @@ public void transformForBqSchema(Map<String, Object> parent, List<Field> bqField
       Map<String, Object> additionalProperties) {
 
     // Clean the key names.
-    ImmutableSet.copyOf(parent.keySet())
-        .forEach(key -> parent.put(getAndCacheBqName(key), parent.remove(key)));
+    final Map<String, String> jsonFieldNames = new HashMap<>();
+    ImmutableSet.copyOf(parent.keySet()).forEach(key -> {
+      String transformedKey = getAndCacheBqName(key);
+      jsonFieldNames.put(transformedKey, key);
```

Comment:
It's technically possible for a name collision to occur in separate branches of a document e.g. `environment.someId` and  `payload.Some_ID`. I think the chances of seeing this in the wild are low, especially considering that this only affects additional_properties in a single document.

Reply:
A logging statement could be one way of warning about a name collision, although it only needs to be emitted as the field is inserted into additional properties. It might also be possible to push the renaming logic closer to insertion into the bq field to avoid this case.

Reply:
This shouldn't actually be an issue, because `jsonFieldNames` isn't passed recursively, so `environment.someId` and `payload.Some_ID` would have different local maps for `jsonFieldNames`.

Regardless, I think it would be better to build `additionalProperties` within this loop, instead of a separate loop, as per [this comment](https://github.com/mozilla/gcp-ingestion/pull/713/files#r308373476)

Reply:
You're right, it looks like name collisions across different namespaces is not an issue.